nle/env/base.py

```diff
@@ -481,147 +268,159 @@ def step(self, action: int):
+        #     ttyrec=self.env._process.filename,
+        # )
+
+    def _in_moveloop(self, observation):
+        program_state = observation[self._program_state_index]
+        return program_state[3]  # in_moveloop
 
     def reset(self):
         """Resets the environment.
 
         Note:
             We attempt to manually navigate the first few menus so that the
             first seen state is ready to be acted upon by the user. This might
-            fail in case NetHack is initialized with some uncommon options.
+            fail in case Nethack is initialized with some uncommon options.
 
         Returns:
-            (dict): observation of the state as defined by
-                    ``self.observation_space``
+            [dict] Observation of the state as defined by
+                `self.observation_space`.
         """
-        self.response = self.env.reset()
+        self._episode += 1
+        new_ttyrec = self._ttyrec_pattern % self._episode if self.savedir else None
+        self.last_observation = self.env.reset(new_ttyrec)
 
         # Only run on the first reset to initialize stats file
         if self._setup_statsfile:
-            stats_file = os.path.splitext(self.env._archive.filename)[0] + ".csv"
-            add_header = not os.path.exists(stats_file)
+            filename = os.path.join(self.savedir, "stats.csv")
+            add_header = not os.path.exists(filename)
 
-            self._stats_file = open(stats_file, "a", 1)  # line buffered
+            self._stats_file = open(filename, "a", 1)  # line buffered.
             self._stats_logger = csv.DictWriter(
                 self._stats_file, fieldnames=self.Stats._fields
             )
             if add_header:
                 self._stats_logger.writeheader()
         self._setup_statsfile = False
 
-        self._episode += 1
-        self._killer_name = "UNK"
+        # self._killer_name = "UNK"
 
         self._steps = 0
-        self._info = self.env._info.copy()
 
-        self._info["seeds"] = {}
-        for k in nethack.SEED_KEYS:
-            self._info["seeds"][k] = getattr(self.response.Seeds(), k.capitalize())()
-
-        while not _in_moveloop(self.response):
+        for _ in range(1000):
             # Get past initial phase of game. This should make sure
             # all the observations are present.
-            self.response, done, _ = self.env.step(ASCII_SPACE)
+            if self._in_moveloop(self.last_observation):
+                break
+            # This fails if the agent picks up a scroll of scare
+            # monster at the 0th turn and gets asked to name it.
+            # Hence the defensive iteration above.
+            self.last_observation, done = self.env.step(ASCII_SPACE)
             assert not done, "Game ended unexpectedly"
+        else:
+            warnings.warn(
+                "Not in moveloop after 1000 tries, aborting (ttyrec: %s)." % new_ttyrec
+            )
+            return reset()
```

Comment:
I suspect this is the cause of hackrl/issues/170

Reply:
Thanks! Fixed in 5260b13.